### PR TITLE
feat: add check to not allow use the installer with previously versions

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

It seems not possible ensure that the workflows can work well within backwards. 
The preflight ensure that the script will not possible try to install/update within an script that is a lower release than the version used to install/update previously. 

#### Which issue(s) this PR fixes:

Fixes # [sc-72065]

#### Special notes for your reviewer:

Test made trying to do the update within an old release:
Screenshot 2023-03-18 at 08 17 30

Test made trying to do the update within an upper release:
Screenshot 2023-03-18 at 08 14 38

#### Does this PR introduce a user-facing change?

```release-note
Adds preflight check to ensure that a host will not be updated with previously kURL releases
```

### CODE REVIEW

File: scripts/common/preflights.sh [hunk 1]
```diff
@@ -657,3 +658,21 @@ function bail_if_unsupported_openebs_to_rook_version() {
...
+       local current_kurl_version
+       current_kurl_version=$(kubectl -n kurl get configmap kurl-current-config -o jsonpath='{ .data.kurl-version }')
+       if [ -z "$current_kurl_version" ]; then
+           current_kurl_version=$(kubectl -n kurl get configmap kurl-last-config -o jsonpath='{ .data.kurl-version }')
+       fi
+
+       semverCompare $(echo "$KURL_VERSION" | sed 's/v//g') $(echo "$current_kurl_version" | sed 's/v//g')
+       if [ "$SEMVER_COMPARE_RESULT"  = "-1" ]; then # greater than or equal to 14.2.21
+           logFail "This host was updated or installed using Kubernetes Installer(kURL) release version"
+           logFail "$current_kurl_version which is earlier than the version $KURL_VERSION."
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:673]: in my opinion the variable `current_kurl_version` is confusing as named and should be named `previous_kurl_version`

```suggestion
           logFail "$current_kurl_version which is greater than the version $KURL_VERSION."
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/preflights.sh [hunk 2]
```diff
@@ -657,3 +658,21 @@ function bail_if_unsupported_openebs_to_rook_version() {
...
+           bail "Please, use an kURL release version which is equals or newer than the version used previously"
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:674]:
```suggestion
           logFail "This host was previously installed using the kURL release version $current_kurl_version"
           logFail "which is greater than the current version $KURL_VERSION."
           bail "Please use a kURL release version which is equal to or greater than the version used previously."
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/preflights.sh [hunk 3]
```diff
@@ -657,3 +658,21 @@ function bail_if_unsupported_openebs_to_rook_version() {
     fi
 }
 
+# not allow run the installer/upgrade when kurl version is lower than the previous applied before
+function bail_if_kurl_version_is_lower_previous_config() {
```
Comment (rrpolanco) [inline scripts/common/preflights.sh:662]:
```suggestion
function bail_if_kurl_version_is_lower_than_previous_config() {
```

File: scripts/common/preflights.sh [hunk 4]
```diff
@@ -657,3 +658,21 @@ function bail_if_unsupported_openebs_to_rook_version() {
...
+    if commandExists kubectl; then
+       local current_kurl_version
+       current_kurl_version=$(kubectl -n kurl get configmap kurl-current-config -o jsonpath='{ .data.kurl-version }')
```
Comment (rrpolanco) [inline scripts/common/preflights.sh:665]: There is a helper function that gets you the "current" kurl version: https://github.com/replicatedhq/kURL/blob/a341c55ae58d340007ad818123d7f65db2316dc1/scripts/common/kurl.sh#L25-L27

Comment (emosbaugh) [reply scripts/common/preflights.sh:665]: oh thank you! our goal should be to add as little new code as possible and use utility functions as much as possible as bash is hard...

File: scripts/common/preflights.sh [hunk 5]
```diff
@@ -657,3 +658,21 @@ function bail_if_unsupported_openebs_to_rook_version() {
     fi
 }
 
+# not allow run the installer/upgrade when kurl version is lower than the previous applied before
+function bail_if_kurl_version_is_lower_previous_config() {
+    if commandExists kubectl; then
+       local current_kurl_version
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:664]: in the interest of not losing my previous comment:

in my opinion the variable current_kurl_version is confusing as named and should be named previous_kurl_version

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/preflights.sh (hunk 5 above)
Comment (camilamacedo86) [reply scripts/common/preflights.sh]: It is addressed with your suggestion

File: scripts/common/preflights.sh (hunk 3 above)
Comment (camilamacedo86) [reply scripts/common/preflights.sh]: Done 👍

File: scripts/common/preflights.sh [hunk 6]
```diff
@@ -657,3 +658,22 @@ function bail_if_unsupported_openebs_to_rook_version() {
     fi
 }
 
+# not allow run the installer/upgrade when kurl version is lower than the previous applied before
+function bail_if_kurl_version_is_lower_than_previous_config() {
+    if commandExists kubectl; then
+       local previous_kurl_version
+       previous_kurl_version="$(kurl_get_current_version)"
+       if [ -z "$previous_kurl_version" ]; then
+               previous_kurl_version="$(kurl_get_current_version)"
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply scripts/common/preflights.sh:667]: OH good catcher. Thank you

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**